libpod/kube.go

```diff
@@ -445,11 +445,17 @@ func (p *Pod) podWithContainers(ctx context.Context, containers []*Container, po
 				if define.IsReservedAnnotation(k) || annotations.IsReservedAnnotation(k) {
 					continue
 				}
-				podAnnotations[fmt.Sprintf("%s/%s", k, removeUnderscores(ctr.Name()))] = TruncateKubeAnnotation(v)
+				if !keepAnnotationLength {
+					v = TruncateKubeAnnotation(v)
```

Comment:
To simplify the code a bit I would change `TruncateKubeAnnotation` to accept a second argument that controls whether it should truncate.  The function is only used internally, so we can make it private:
```golang
func truncateKubeAnnotation(value string, truncate bool) string { ... }
```